### PR TITLE
Integration-Test Suite

### PR DESCRIPTION
**Pull request by shurvitz**
**Description of changes:** Integration-Test Suite using CDK to create CloudFormation stack with all resources. The Suite manages the infra (creates/deletes the CF stack) and allows each connector to customize their integration-tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-federation-sdk-tools/pom.xml [hunk 1]
```diff
@@ -70,6 +70,24 @@
             <version>${log4j2Version}</version>
             <scope>runtime</scope>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-cloudformation -->
```
Comment (avirtuos) [inline athena-federation-sdk-tools/pom.xml:73]: This need to be test dependencies so they don't go into the jar that is fed to Lambda

Comment (shurvitz) [reply athena-federation-sdk-tools/pom.xml:73]: changed CloudFormation dependecy scope to provided `<scope>provided</scope>` in sdk-tools pom. Dependency will need to be provided in the specific connectors' pom with a `<scope>test</scope>`

File: athena-federation-sdk/pom.xml [hunk 2]
```diff
@@ -22,7 +22,6 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <slf4jVersion>1.7.30</slf4jVersion>
         <log4j2Version>2.13.3</log4j2Version>
-        <aws-sdk.version>1.11.769</aws-sdk.version>
```
Comment (burhan94) [inline athena-federation-sdk/pom.xml:25]: why was this removed?

Comment (shurvitz) [reply athena-federation-sdk/pom.xml:25]: The variable was duplicated from the main app pom causing a conflict situation where the Federation SDK could have a different AWS SDK version than the Connectors.

**Review by burhan94 — APPROVED** [review]
